### PR TITLE
Add socket diagnosis for UDP

### PR DESCRIPTION
**Pull request by srebhan**
This PR adds netlink socket diagnosis support for UDP and implements parsing of the `INET_DIAG_MEMINFO` extension data.

### CODE REVIEW

**Comment by aboch** [comment]
@srebhan This project follows a "1 commit per PR" approach.
Either squash the commits into one or open a separate PR for each, if each brings a distinct fix/functionality.

**Comment by srebhan** [comment]
@aboch will squash the commits then. Thought you squash them on merge...

File: socket_linux.go [hunk 1]
```diff
@@ -263,29 +344,44 @@ loop:
...
+			if err := info.TCPInfo.deserialize(a.Value); err != nil {
 				return nil, err
 			}
-			continue
+		case INET_DIAG_BBRINFO:
+			info.TCPBBRInfo = &TCPBBRInfo{}
+			if err := info.TCPBBRInfo.deserialize(a.Value); err != nil {
+				return nil, err
+			}
+			// default:
```
Comment (aboch) [inline socket_linux.go:362]: If it makes sense to handle the default case, please uncomment it. If not, please remove it.

File: socket_linux.go [hunk 2]
```diff
@@ -263,29 +344,44 @@ loop:
...
+	}
+	for _, a := range attrs {
+		switch a.Attr.Type {
+		case INET_DIAG_MEMINFO:
+			info.Memory = &MemInfo{}
+			if err := info.Memory.deserialize(a.Value); err != nil {
 				return nil, err
 			}
-			continue
+			// default:
```
Comment (aboch) [inline socket_linux.go:381]: same here

**Comment by aboch** [comment]
Can you see if possible to add or extend existing UT to cover this functionality?

**Comment by srebhan** [comment]
@aboch any update on this PR? Anything I can do to make the merge happen?

File: go.mod [hunk 3]
```diff
@@ -3,6 +3,7 @@ module github.com/vishvananda/netlink
 go 1.12
 
 require (
+	github.com/stretchr/testify v1.8.4
```
Comment (aboch) [inline go.mod:6]: This is not needed for the feature added by this PR.
Can you please remove?

File: socket_test.go [hunk 4]
```diff
@@ -1,77 +1,65 @@
...
 )
 
 func TestSocketGet(t *testing.T) {
 	defer setUpNetlinkTestWithLoopback(t)()
 
 	addr, err := net.ResolveTCPAddr("tcp", "localhost:0")
-	if err != nil {
-		log.Fatal(err)
-	}
+	require.NoError(t, err)
```
Comment (aboch) [inline socket_test.go:20]: No need for this for this feature addition.
Existing testing code is readable just fine IMO.

**Review by aboch — CHANGES_REQUESTED** [review]

**Comment by aboch** [comment]
please squash the two commits into one

**Comment by srebhan** [comment]
@aboch done.

**Pull request merged**